### ISSUE TITLE
[Request]: Interface to out-of-game files

### ISSUE THREAD

**Issue by jjclark1982** [Turing-Complete-Game/Suggestions-and-Issues#173]
### Request?

I would like to be able to read inputs from a text file, and send outputs to another text file.

For example, I could place `inputs.txt` in the `test_lab` folder, containing a series of numbers:
```
10
20
30
40
50
```
And running the program would produce an `output.txt` file containing the output numbers.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by EnderShadow** [comment]
I think a better option would be a way to set the list of inputs and the expected outputs. Currently this can be done at a basic level with set_input and expect, but since that's entirely controlled by the program a more in-depth way would be better. This could probably be part of user-made levels as well when they are added.

**Comment by jjclark1982** [comment]
Yes, it would be much more flexible if `set_input` could put values into a queue instead of just overwriting the current value. Then the assembly code could include the whole sequence, even if it does not match the program's control flow structure.

**Comment by EnderShadow** [comment]
I know this isn't an optimal solution, but in the meantime you could use a program component with a counter and you can just set it so that it only increments when you read from input. You might have to handle hitting the end of program memory though. (although you could use the 64 bit memory to push that point to 2^16

**Comment by Stuffe** [comment]
Is this for testing? There will be file IO components soon

**Comment by SunCatMC** [comment]
FileROM for reading from file exists now, but writing to file is not yet possible

**Comment by Stuffe** [comment]
File read is added, file write will be here soon. Closing now so I don't have to go through it again
